src/OrchardCore.Modules/OrchardCore.Contents/Services/DefaultContentsAdminListQueryService.cs

```diff
@@ -1,45 +1,128 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using OrchardCore.ContentManagement;
 using OrchardCore.Contents.ViewModels;
 using OrchardCore.DisplayManagement.ModelBinding;
 using OrchardCore.Modules;
 using YesSql;
+using YesSql.Filters.Abstractions.Nodes;
 
 namespace OrchardCore.Contents.Services
 {
     public class DefaultContentsAdminListQueryService : IContentsAdminListQueryService
     {
+        private readonly static List<string> _operators = new()
+        {
+            "OR", "AND", "||", "&&"
+        };
+
         private readonly ISession _session;
         private readonly IServiceProvider _serviceProvider;
         private readonly IEnumerable<IContentsAdminListFilter> _contentsAdminListFilters;
+        private readonly ContentsAdminListFilterOptions _contentsAdminListFilterOptions;
         private readonly ILogger _logger;
 
         public DefaultContentsAdminListQueryService(
             ISession session,
             IServiceProvider serviceProvider,
             IEnumerable<IContentsAdminListFilter> contentsAdminListFilters,
+            IOptions<ContentsAdminListFilterOptions> contentsAdminListFilterOptions,
             ILogger<DefaultContentsAdminListQueryService> logger)
         {
             _session = session;
             _serviceProvider = serviceProvider;
             _contentsAdminListFilters = contentsAdminListFilters;
+            _contentsAdminListFilterOptions = contentsAdminListFilterOptions.Value;
             _logger = logger;
         }
 
         public async Task<IQuery<ContentItem>> QueryAsync(ContentOptionsViewModel model, IUpdateModel updater)
         {
-            // Because admin filters can add a different index to the query this must be added as a Query<ContentItem>()
+            var defaultTermNode = model.FilterResult.OfType<DefaultTermNode>().FirstOrDefault();
+            var defaultTermName = defaultTermNode?.TermName;
+            var defaultOperator = defaultTermNode?.Operation;
+            var shouldRestoreDefault = false;
+
+            if (defaultTermNode is not null)
+            {
+                var value = defaultTermNode.ToString();
+                if (!_operators.Any(opt => value.Contains(opt, StringComparison.Ordinal)))
+                {
+                    // Use an unary operator based on a full quoted string.
+                    defaultOperator = new UnaryNode(value, OperateNodeQuotes.Double);
+                }
+
+                var selectedContentType = GetSelectedContentType(model);
+
+                if (selectedContentType != null)
```

Comment:
Maybe `is not null`, and no blank line before, that's a new habit I have ;)

Reply:
Why are we used `is not null` now instead of `!= null`

Reply:
Not so recent but it's a new c# feature, belonging to pattern matching if I remember correctly